eos/utils/expression-parser_TEST.cc

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2021 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <test/test.hh>
+
+#include <eos/utils/expression.hh>
+#include <eos/utils/expression-fwd.hh>
+#include <eos/utils/expression-parser-impl.hh>
+#include <eos/utils/expression-visitors.cc>
+#include <eos/utils/kinematic.hh>
+#include <eos/utils/options.hh>
+#include <eos/utils/parameters.hh>
+
+#include <iostream>
+
+using namespace test;
+using namespace eos::exp;
+using namespace eos;
+
+class ExpressionTest
+{
+    private:
+        const std::string _input;
+
+    public:
+        Expression e;
+        bool completed;
+
+        using It = std::string::const_iterator;
+        ExpressionParser<It> parser;
+
+        ExpressionTest(const std::string input) :
+            _input(input)
+        {
+            It first(_input.begin()), last(_input.end());
+            completed = qi::phrase_parse(first, last, parser, ascii::space, e) && (first == last);
+        }
+};
+
+
+class ExpressionParserTest :
+    public TestCase
+{
+public:
+    ExpressionParserTest() :
+    TestCase("expression_parser_test")
```

Comment:
Coding style: initalizations/constructor calls are indented one level (4 spaces) with respect to the class name.